autometa/binning/recursive_dbscan.py

```diff
@@ -91,16 +93,18 @@ def recursive_dbscan(table, markers_df, domain, completeness_cutoff, purity_cuto
     """
     eps = 0.3
     step = 0.1
-    clustering_rounds = {0:0}
+    clustering_rounds = {0: 0}
     n_clusters = float('inf')
     best_median = float('-inf')
     best_df = pd.DataFrame()
     while n_clusters > 1:
         binned_df = run_dbscan(table, eps, method=method)
-        df = add_metrics(df=binned_df, markers_df=markers_df, domain=domain)
-        completess_filter = df['completeness'] >= completeness_cutoff
-        purity_filter = df['purity'] >= purity_cutoff
-        median_completeness = df[completess_filter & purity_filter]['completeness'].median()
+        df, metrics_df = add_metrics(
+            df=binned_df, markers_df=markers_df, domain=domain)
+        completeness_filter = metrics_df['completeness'] >= completeness_cutoff
+        purity_filter = metrics_df['purity'] >= purity_cutoff
+        median_completeness = metrics_df[completeness_filter &
+                                         purity_filter]['completeness'].median()
```

Comment:
This doesn't use groupby methods like in your comment on issue #52. So will this calculate the median completeness of each bin?

Reply:
Yes,

```python
# metrics_df is indexed by cluster.
>>>metrics_df[completeness_filter & purity_filter].head()
       purity  completeness
0     96.4286     20.143885
1         100     27.338129
5     96.3636     39.568345
6         100     22.302158
7     97.7528     64.028777

>>>metrics_df[completeness_filter & purity_filter].median()
purity          100.000000
completeness     27.338129
dtype: float64

>>>metrics_df[completeness_filter & purity_filter]["completeness"].median()
27.33812949640288

```